library/network/src/modules/SuSEFirewall.rb

```diff
@@ -154,9 +154,64 @@ def self.create
   # ----------------------------------------------------------------------------
   # SuSEFirewalld Class. Trying to provide relevent pieces of SF2 functionality via
   # firewalld.
-  class SuSEFirewalld < Module
+  class SuSEFirewalld < Firewall
+    attr_reader :special_all_interface_zone
+
+    # Valid attributes for firewalld zones
```

Comment:
I see. As you see, it is not obvious, so it should be written in a comment attached to where the things are declared, probably right here.

`:modified` is implemented with an array, but if you use [Set from the standard library](http://ruby-doc.org/stdlib-2.0.0/libdoc/set/rdoc/Set.html) you can stop worrying about `uniq!`.